multiplex_immuno_processing/execute_round_alignment.py

```diff
@@ -0,0 +1,199 @@
+import argparse
+from glob import glob
+import os
+from pathlib import Path
+import re
+import registration_utils
+from aicsimageio import AICSImage
+import numpy as np
+import pandas as pd
+from scipy.ndimage import affine_transform
+import skimage.io as skio
+import yaml
+from yaml.loader import SafeLoader
+import tqdm
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "--dfall_csv_dir",
+    type=str,
+    required=True,
+    help="csv file",
+)
+
+parser.add_argument(
+    "--barcode", type=str, required=True, help="specify barcode to analyze"
+)
+
+parser.add_argument(
+    "--output_path",
+    type=str,
+    required=True,
+    help="output dir of all processing steps. This specifies where to find the yml_configs too",
+)
+
+parser.add_argument(
+    "--position",
+    type=str,
+    required=True,
+    help="position"
+)
+
+parser.add_argument("--method", choices=['cross_cor', 'ORB'])
+
+
+if __name__ == "__main__":
+    args = parser.parse_args()
+    dfall = pd.read_csv(args.dfall_csv_dir)
+    mag = "20x"
+
+    Position = args.position
+
+    yaml_dir = os.path.join(args.output_path, "yml_configs")
+    barcode = args.barcode
+
+    yaml_list = [x for x in os.listdir(yaml_dir) if "_confirmed" in x]
+    yaml_list = [x for x in yaml_list if barcode in x]
+    dflist = []
+    for y in yaml_list:
+        print(y)
+        yml_path = yaml_dir + os.sep + y
+        with open(yml_path) as f:
+            data = yaml.load(f, Loader=SafeLoader)
+            for round_dict in data["Data"]:
+                dfsub = pd.DataFrame(round_dict.values(), index=round_dict.keys()).T
+                dfsub["barcode"] = data["barcode"]
+                dfsub["scope"] = data["scope"]
+                dfsub["output_path"] = data["output_path"]
+                dflist.append(dfsub)
+
+        dfconfig = pd.concat(dflist)
+        dfconfig.set_index(["barcode", "round"], inplace=True)
+
+
+
+
+
+    keylist = dfall.set_index("template_position").loc[args.position, "key"].unique()
```

Comment:
this only aligns one position. I thought we decided to make it align all positions unless a single position was specified. So the -position argument could accept 1. a single position, 2. a list of positions or 3. no argument (in which case all positions would be aligned)

Reply:
Look into keep this general enough where one can run this for all positions without jinja setup